---
title: "REGENIE UKB_BC"
output: html_document
date: "2025-10-30"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


Regenie step 1 prep using genotype data
```{r}
#Merge all genotype calls of each chromosome into one
#File path: project-GypBPq0Jf7kBGqKGJqVKBjg3:/Bulk/Genotype Results/Genotype calls/
#plink
rm -f list_beds.txt
for chr in {2..22}; do echo "ukb22418_c${chr}_b0_v2.bed ukb22418_c${chr}_b0_v2.bim ukb22418_c${chr}_b0_v2.fam" >> list_beds.txt; done #creating a list with only chr2-22

plink \
  --bed ukb22418_c1_b0_v2.bed \
  --bim ukb22418_c1_b0_v2.bim \
  --fam ukb22418_c1_b0_v2.fam \
  --merge-list list_beds.txt \
  --make-bed --out ukb_cal_allChrs #merging that list with chr1

#one line version for DNAnexus

rm -f list_beds.txt; for chr in {2..22}; do echo "ukb22418_c${chr}_b0_v2.bed ukb22418_c${chr}_b0_v2.bim ukb22418_c${chr}_b0_v2.fam" >> list_beds.txt; done; plink --bed ukb22418_c1_b0_v2.bed --bim ukb22418_c1_b0_v2.bim --fam ukb22418_c1_b0_v2.fam --merge-list list_beds.txt --make-bed --out ukb_cal_allChrs

#qc 
#have to pick a HWE threshold
#Discussion on HWE threshold on this analysis
#1e-6 seems to be innapropriate for a large sample like the UKB 
#find out the sample size and give reasoning on why you used 1e-15 0.01
#k parameter is recommended by Greer PJ, et al. (2024)

#the snp_failed.. waws conducted based on array batch qc provided by UKB
#https://biobank.ndph.ox.ac.uk/ukb/refer.cgi?id=1955

plink2 \
  --bfile ukb_cal_allChrs \
  --exclude snp_failed_arraybatch_qc.txt
  --maf 0.01 --mac 100 --geno 0.01 --hwe 1e-15 0.001 midp \
  --mind 0.1 \
  --make-bed \
  --out qc_pass

#oneline version for dnanexus

plink2 --bfile ukb_cal_allChrs --exclude snp_failed_arraybatch_qc.txt --maf 0.01 --mac 100 --geno 0.01 --hwe 1e-15 0.001 midp --mind 0.1 --make-bed --out qc_pass
```

The UK Biobank did the filtering itself for qc, so might be doing too much fltering by usin the default UKB qc list plus our own filtering with maf mac geno hwe mind etc etc 

Need to include LD filtering 
```{r}
#LD filtering 
#make list of snps passing and not passing the LD requirements
plink2 --bfile qc_pass --indep-pairwise 1000 100 0.9 --out qc_pass_ld

#keep the SNPs passing the LD pruning
plink2 --bfile qc_pass --extract qc_pass_ld.prune.in --make-bed --out qc_pass_ld

## took too long so this is the alternative based on joe's code
plink2 --memory 8000 --bfile qc_pass --indep-pairwise 1000 100 0.8 --out qc_pass_ld08 --threads 6 && \ plink2 --memory 1000 --bfile qc_pass --extract qc_pass_ld08.prune.in --make-bed --out qc_pass_ld08
```

Needed to also filter out samples excluded based on sex, kinship, ancestry from UKB provided files - maybe not then
```{r}
if("librarian" %in% installed.packages()==FALSE) install.packages("librarian")
librarian::shelf(tidyverse)


status <- read.table("./data/raw/bc.txt", header = TRUE)
remove_qc <- read.table("./data/raw/fid_to_remove_breast.txt", header = TRUE)

# remove all ids from covariate and from status 

filter_status <- anti_join(status, remove_qc, by = "FID")
#197602 final number 

sum(is.na(filter_cov))
sum(is.na(filter_status))

#upload filtered datasets
write.table(filter_status, file = "./data/processed/filter_bc.txt", sep = "\t", row.names = FALSE, quote = FALSE)
```


starting with regenie step 1
```{r}
./regenie \
  --step 1 \
  --bed qc_pass_ld08 \
  --phenoFile bc.txt \
  --pheoColList ... # i think we can have more than one phenotype column here?
  --remove fid_to_remove_breast.txt \
  --covarFile covariates_breast.txt \
  --covarColList ... # specify which covariates to include #age?? #PC1-10 for sure
  --bt \
  --bsize 1000 \
  --lowmem \
  --lowmem-prefix tmpdir/regenie_tmp_preds \
  --threads 8 \
  --out ukb_step1_BT
  --loocv 

#one line version - modify so it corresponds to the above one
regenie --step 1 --bed ukb_cal_allChrs --bed qc_pass_ld08 --phenoFile bc.txt --remove fid_to_remove_breast.txt --covarFile covariates_breast.txt --bt --bsize 1000 --lowmem --lowmem-prefix tmpdir/regenie_tmp_preds --out ukb_step1_BT
```


```{r}
./regenie \
  --step 2 \
  --bgen ukb22418_c22_b0_v2.bgen \
  --ref-first \
  --sample ukb22418_c22_b0_v2.sample \
  --phenoFile filter_bc.txt \
  --phenoColList
  --covarFile filter_cov_bc.txt \
  --covarCollist 
  --pred ukb_step1_BT_pred.list \
  --bt \
  --firth --approx --pThresh 0.01 \ # need to insert our own p value threshold 
  --bsize 400 \
  --threads 8 \
  --af-cc	#for output format case/control
  --out ukb_step2_BT_chr22

##To remove all samples that have missing values at any of the P
 ##phenotypes, use option --strict

#one line version
```

