---
title: "REGENIE UKB_BC"
output: html_document
date: "2025-10-30"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R 
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


Regenie step 1 prep using genotype data
```{r}
#Merge all genotype calls of each chromosome into one
#File path: project-GypBPq0Jf7kBGqKGJqVKBjg3:/Bulk/Genotype Results/Genotype calls/
#plink
rm -f list_beds.txt
for chr in {2..22}; do echo "ukb22418_c${chr}_b0_v2.bed ukb22418_c${chr}_b0_v2.bim ukb22418_c${chr}_b0_v2.fam" >> list_beds.txt; done #creating a list with only chr2-22

plink \
  --bed ukb22418_c1_b0_v2.bed \
  --bim ukb22418_c1_b0_v2.bim \
  --fam ukb22418_c1_b0_v2.fam \
  --merge-list list_beds.txt \
  --make-bed --out ukb_cal_allChrs #merging that list with chr1

#one line version for DNAnexus

rm -f list_beds.txt; for chr in {2..22}; do echo "ukb22418_c${chr}_b0_v2.bed ukb22418_c${chr}_b0_v2.bim ukb22418_c${chr}_b0_v2.fam" >> list_beds.txt; done; plink --bed ukb22418_c1_b0_v2.bed --bim ukb22418_c1_b0_v2.bim --fam ukb22418_c1_b0_v2.fam --merge-list list_beds.txt --make-bed --out ukb_cal_allChrs

#qc 
#have to pick a HWE threshold
#Discussion on HWE threshold on this analysis
#1e-6 seems to be innapropriate for a large sample like the UKB 
#find out the sample size and give reasoning on why you used 1e-15 0.01
#k parameter is recommended by Greer PJ, et al. (2024)

#the snp_failed.. waws conducted based on array batch qc provided by UKB
#https://biobank.ndph.ox.ac.uk/ukb/refer.cgi?id=1955

plink2 \
  --bfile ukb_cal_allChrs \
  --exclude snp_failed_arraybatch_qc.txt
  --maf 0.01 --mac 100 --geno 0.01 --hwe 1e-15 0.001 midp \
  --mind 0.1 \
  --make-bed \
  --out qc_pass

#oneline version for dnanexus

plink2 --bfile ukb_cal_allChrs --exclude snp_failed_arraybatch_qc.txt --maf 0.01 --mac 100 --geno 0.01 --hwe 1e-15 0.001 midp --mind 0.1 --make-bed --out qc_pass
```

The UK Biobank did the filtering itself for qc, so might be doing too much fltering by usin the default UKB qc list plus our own filtering with maf mac geno hwe mind etc etc 

Need to include LD filtering 
```{r}
#LD filtering 
#make list of snps passing and not passing the LD requirements

#LD pruning takes a long tiem with all aprticipants, so made a new file with only a subset of random 10k participants

plink2 --memory 8000 --bfile qc_pass --keep id_for_ld.txt --indep-pairwise 1000 100 0.9 --out qc_pass_ld09 --threads 6
```

starting with regenie step 1 - breast
```{r}
mkdir -p tmpdir && regenie --step 1 --bed qc_pass --extract qc_pass_ld09_full.prune.in --remove fid_to_remove_breast.txt --phenoFile pheno_breast.txt --phenoColList bbd_pl_pros,bbd_npl_pros,bbd_bn_pros,dcis_pros,lcis_pros,bc --covarFile covariates.txt --covarColList PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --bt --bsize 1000 --lowmem --lowmem-prefix tmpdir/regenie_tmp_preds --out ukb_step1_BT --loocv

```

regenie step 1 - lung (smoker)
```{r}
mkdir -p tmpdir && regenie --step 1 --bed qc_pass --extract qc_pass_ld09_full.prune.in --remove fid_to_remove_lung.txt --phenoFile lung_smoker.txt --phenoColList nodules_pros,lc --covarFile covariates.txt --covarColList PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --bt --bsize 1000 --lowmem --lowmem-prefix tmpdir/regenie_tmp_preds --out ukb_step1_BT_lung_smoker --loocv
```

regenie step 1 - lung (nonsmoker)
```{r}
mkdir -p tmpdir && regenie --step 1 --bed qc_pass --extract qc_pass_ld09_full.prune.in --remove fid_to_remove_lung.txt --phenoFile lung_neversmoke.txt --phenoColList nodules_pros,lc --covarFile covariates.txt --covarColList PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --bt --bsize 1000 --lowmem --lowmem-prefix tmpdir/regenie_tmp_preds --out ukb_step1_BT_lung_never --loocv
```

