---
title: "post-gwas"
author: "Christis Nicolaides"
date: "2025-12-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
```{r}

bc <- read.table("./data/raw/bc.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
dcis <- read.table("./data/raw/dcis.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lcis <- read.table("./data/raw/lcis.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_pl <- read.table("./data/raw/bbd_pl.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_npl <- read.table("./data/raw/bbd_npl.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_bn <- read.table("./data/raw/bbd_bn.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lc_smoker <- read.table("./data/raw/lc_smoker.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lc_neversmoke <- read.table("./data/raw/lc_neversmoke.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
nodules_smoker <- read.table("./data/raw/nodules_smoker.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
nodules_neversmoke <- read.table("./data/raw/nodules_neversmoke.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)

```


create a column in this format 
10:121586676:A:G	

```{r}
#Download necessary packages

library(tidyverse)

if("librarian" %in% installed.packages()==FALSE) install.packages("librarian")
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(c(
  "GenomicRanges",
  "TxDb.Hsapiens.UCSC.hg38.knownGene",
  "org.Hs.eg.db"
))

librarian::shelf(dplyr, ggplot2, tidyr, readr, purrr, GenomicRanges, TxDb.Hsapiens.UCSC.hg38.knownGene, org.Hs.eg.db, RCytoGPS, snpsettest)


```

```{r}
# new columns 

datasets <- c(
  "bc", 
  "lcis", 
  "dcis", 
  "bbd_bn", 
  "bbd_npl", 
  "bbd_pl", 
  "lc_smoker", 
  "lc_neversmoke",
  "nodules_smoker",
  "nodules_neversmoke"
)

for (d in datasets) {
  df <- get(d) %>% 
    mutate(
      variant_id = str_c(CHROM, GENPOS, ALLELE0, ALLELE1, sep = ":")
    )
  
  assign(d, df)
}

```

VEP annotated datatsets 

```{r}
bc_vep <- read_tsv(
  "./data/raw/bc_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

lcis_vep <- read_tsv(
  "./data/raw/lcis_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

dcis_vep <- read_tsv(
  "./data/raw/dcis_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

bbd_bn_vep <- read_tsv(
  "./data/raw/bbd_bn_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

bbd_pl_vep <- read_tsv(
  "./data/raw/bbd_pl_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

bbd_npl_vep <- read_tsv(
  "./data/raw/bbd_npl_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)


lc_neversmoke_vep <- read_tsv(
  "./data/raw/lc_neversmoke_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

lc_smoker_vep <- read_tsv(
  "./data/raw/lc_smoker_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

nodules_neversmoke_vep <- read_tsv(
  "./data/raw/nodules_neversmoke_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

nodules_smoker_vep <- read_tsv(
  "./data/raw/nodules_smoker_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)
#delete unessesary columns to save memory

bc_vep <- bc_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

lcis_vep <- lcis_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

dcis_vep <- dcis_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

bbd_bn_vep <- bbd_bn_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

bbd_npl_vep <- bbd_npl_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

bbd_pl_vep <- bbd_pl_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

lc_neversmoke_vep <- lc_neversmoke_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

lc_smoker_vep <- lc_smoker_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

nodules_neversmoke_vep <- nodules_neversmoke_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

nodules_smoker_vep <- lc_smoker_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)


#create a k
```

```{r}
#make sure whether all variants in bc are in bc_vep

missing_variants <- setdiff(bc$variant_id, bc_vep$`#Uploaded_variation`) #empty

#all represented 

#create a new column on bc which cas the gene symbol based on the data of bc_vep

bc <- bc %>%
  left_join(
    bc_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

lcis <- lcis %>%
  left_join(
    lcis_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

dcis <- dcis %>%
  left_join(
    dcis_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  )%>%
  unique()


lc_neversmoke <- lc_neversmoke %>%
  left_join(
    lc_neversmoke_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  )%>%
  unique()

bbd_bn <- bbd_bn %>%
  left_join(
    bbd_bn_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

bbd_pl <- bbd_pl %>%
  left_join(
    bbd_pl_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

bbd_npl <- bbd_npl %>%
  left_join(
    bbd_npl_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

lc_smoker <- lc_smoker %>%
  left_join(
    lc_smoker_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

lc_neversmoke <- lc_neversmoke %>%
  left_join(
    lc_neversmoke_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

nodules_neversmoke <- nodules_neversmoke %>%
  left_join(
    nodules_neversmoke_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

nodules_smoker <- nodules_smoker %>%
  left_join(
    nodules_smoker_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()
```


```{r}
#unique geens affected in each phenotype 
bc_clean                 <- bc %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
dcis_clean               <- dcis %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
lcis_clean               <- lcis %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
bbd_pl_clean             <- bbd_pl %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
bbd_npl_clean            <- bbd_npl %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
bbd_bn_clean             <- bbd_bn %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
lc_smoker_clean          <- lc_smoker %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
lc_neversmoke_clean      <- lc_neversmoke %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
nodules_smoker_clean     <- nodules_smoker %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))
nodules_neversmoke_clean <- nodules_neversmoke %>% filter(EXTRA != "TEST_FAIL" | is.na(EXTRA))

bc_genes <- unique(na.omit(bc_clean$SYMBOL))
lcis_genes <- unique(na.omit(lcis_clean$SYMBOL))
dcis_genes <- unique(na.omit(dcis_clean$SYMBOL))
bbd_bn_genes <- unique(na.omit(bbd_bn_clean$SYMBOL))
bbd_pl_genes <- unique(na.omit(bbd_pl_clean$SYMBOL))
bbd_npl_genes <- unique(na.omit(bbd_npl_clean$SYMBOL))
lc_smoker_genes <- unique(na.omit(lc_smoker_clean$SYMBOL))
lc_neversmoke_genes <- unique(na.omit(lc_neversmoke_clean$SYMBOL))
nodules_smoker_genes <- unique(na.omit(nodules_smoker_clean$SYMBOL))
nodules_neversmoke_genes <- unique(na.omit(nodules_neversmoke_clean$SYMBOL))

#make a table with the column being the phenotype and the rows being the genes
gene_list <- list(
  bc = bc_genes,
  lcis = lcis_genes,
  dcis = dcis_genes,
  bbd_bn = bbd_bn_genes,
  bbd_pl = bbd_pl_genes,
  bbd_npl = bbd_npl_genes,
  lc_smoker = lc_smoker_genes,
  lc_neversmoke = lc_neversmoke_genes,
  nodules_smoker = nodules_smoker_genes,
  nodules_neversmoke = nodules_neversmoke_genes
)

```


```{r}
#for the phenotypes of lcis, dcis, bbbd_bn, bbd_pl, bbd_npl, lc_smoker, lc_neversmoke, nodules_smoker, nodules_neversmoke make a plot with the NORMAL distribution of allele frequency A1FREQ column only for the variants which TEST_FAIL in the EXTRA column 

library(dplyr)
library(ggplot2)
library(purrr)

datasets <- list(
  lcis = lcis,
  dcis = dcis,
  bbd_bn = bbd_bn,
  bbd_pl = bbd_pl,
  bbd_npl = bbd_npl,
  lc_smoker = lc_smoker,
  lc_neversmoke = lc_neversmoke,
  nodules_smoker = nodules_smoker,
  nodules_neversmoke = nodules_neversmoke
)

# Function to filter TEST_FAIL and plot A1FREQ distribution
plot_A1freq_distribution <- function(data, name) {
  
  df <- data %>%
    filter(EXTRA == "TEST_FAIL") %>%
    filter(!is.na(A1FREQ))
  
  if (nrow(df) == 0) {
    message(paste("No TEST_FAIL variants for:", name))
    return(NULL)
  }
  
  ggplot(df, aes(x = A1FREQ)) +
    geom_histogram(aes(y = ..density..), bins = 40, alpha = 0.5) +
    geom_density(linewidth = 1.2) +
    labs(
      title = paste("Normal Distribution of A1FREQ for TEST_FAIL variants:", name),
      x = "Allele Frequency (A1FREQ)",
      y = "Density"
    ) +
    theme_minimal()
}

plots <- imap(datasets, plot_A1freq_distribution)
# Display plots

plots

```


```{r}
#Rcyto and gene mapping
prep_for_mapping <- function(df) {
  df %>%
    mutate(
      # make a unique SNP ID per row
      snp = paste0(variant_id, "_", dplyr::row_number()),
      chr = as.character(CHROM),
      chr = dplyr::case_when(
        chr %in% c("X", "23") ~ "23",
        chr %in% c("Y", "24") ~ "24",
        TRUE ~ chr
      ),
      chr = as.numeric(chr),
      pos = GENPOS
    )
}


datasets <- list(
  bc_clean                 = bc_clean,
  lcis_clean               = lcis_clean,
  dcis_clean               = dcis_clean,
  bbd_bn_clean             = bbd_bn_clean,
  bbd_pl_clean             = bbd_pl_clean,
  bbd_npl_clean            = bbd_npl_clean,
  lc_neversmoke_clean      = lc_neversmoke_clean,
  lc_smoker_clean          = lc_smoker_clean,
  nodules_neversmoke_clean = nodules_neversmoke_clean,
  nodules_smoker_clean     = nodules_smoker_clean
)


###template from Dr Daniel Barnes####
map.cyto.gene <- function(d,build=38,sta=20000,end=20000,...) {
  require(RCytoGPS) # for mapping variant to cytoband using data(cytobandLocations)
  require(snpsettest) # for mapping variant to gene with "map_snp_to_gene" command
  
  if(!("snp"%in%colnames(d))) { stop("Column for SNP name (snp) not present") }
  if(!("chr"%in%colnames(d))) { stop("Column for chromosome (chr) not present") }
  if(!("pos"%in%colnames(d))) { stop("Column for position (pos) not present") }
  if(!(build)%in%c(37,38)) { stop("Human genome build must take values 37 (GRCh37) or 38 (GRCh38)") }
  if(sta<0) { stop("Distance before the gene must be a non-negative integer") }
  if(end<0) { stop("Distance after the gene must be a non-negative integer") }
  
  #############
  # Gene data #
  #############
  x <- d[,c("snp","chr","pos")]
  colnames(x)[1] <- "id"
  sta <- sta/1000
  end <- end/1000
  if(build==37) {
    g <- gene.curated.GRCh37
  }
  if(build==38) {
    g <- gene.curated.GRCh38
  }
  x <- map_snp_to_gene(info_snp=x,g,extend_start=sta,extend_end=end,only_sets=FALSE)
  g <- g[which(g$gene.id%in%x$map$gene.id),]
  d$gene <- ""
  for(i in 1:nrow(d)) {
    snp <- d$snp[i]
    map <- x$map[which(x$map==snp),]
    if(is.na(mean(map$gene.start))) {
      # do nothing
    } else {
      gene.ids <- unique(map$gene.id)
      genes <- g[which(g$gene.id%in%gene.ids),]$gene.name # match gene.ids to "g"
      d[i,]$gene <- paste(genes,collapse=", ")
      rm(gene.ids,genes)
    }
    rm(snp,map)
  }; rm(i)
  
  # Cytoband data: cytoband locations were unchanged from b37 to b38
  map <- cytobandLocations
  map$chr <- gsub("chr","",map$Chromosome,fixed=TRUE)
  map[which(map$chr=="X"),]$chr <- "23"
  map[which(map$chr=="Y"),]$chr <- "24"
  
  map$chr <- as.numeric(map$chr)
  d$cytoband <- ""
  for(i in 1:nrow(d)) {
    chr <- d$chr[i]
    pos <- d$pos[i]
    cyt <- row.names(map[which(map$chr==chr & (map$loc.start<=pos & map$loc.end>=pos)),])
    d[i,]$cytoband <- cyt
    rm(chr,pos,cyt)
  }; rm(i)
  
  return(d)
}

data("cytobandLocations", package = "RCytoGPS")
data("gene.curated.GRCh38", package = "snpsettest") 

mapped <- lapply(datasets, function(df) {
  df_prepped <- prep_for_mapping(df)
  map.cyto.gene(df_prepped, build = 38, sta = 20000, end = 20000)
})

bc_clean                 <- mapped$bc_clean
lcis_clean               <- mapped$lcis_clean
dcis_clean               <- mapped$dcis_clean
bbd_bn_clean             <- mapped$bbd_bn_clean
bbd_pl_clean             <- mapped$bbd_pl_clean
bbd_npl_clean            <- mapped$bbd_npl_clean
lc_neversmoke_clean      <- mapped$lc_neversmoke_clean
lc_smoker_clean          <- mapped$lc_smoker_clean
nodules_neversmoke_clean <- mapped$nodules_neversmoke_clean
nodules_smoker_clean     <- mapped$nodules_smoker_clean

```


```{r}
#modify the gene list to include all unique genes from the SYMBOL column but now also from the gene column
bc_genes <- unique(na.omit(c(bc_clean$SYMBOL, bc_clean$gene)))
lcis_genes <- unique(na.omit(c(lcis_clean$SYMBOL, lcis_clean$gene)))
dcis_genes <- unique(na.omit(c(dcis_clean$SYMBOL, dcis_clean$gene)))
bbd_bn_genes <- unique(na.omit(c(bbd_bn_clean$SYMBOL, bbd_bn_clean$gene)))
bbd_pl_genes <- unique(na.omit(c(bbd_pl_clean$SYMBOL, bbd_pl_clean$gene)))
bbd_npl_genes <- unique(na.omit(c(bbd_npl_clean$SYMBOL, bbd_npl_clean$gene)))
lc_smoker_genes <- unique(na.omit(c(lc_smoker_clean$SYMBOL, lc_smoker_clean$gene)))
lc_neversmoke_genes <- unique(na.omit(c(lc_neversmoke_clean$SYMBOL, lc_neversmoke_clean$gene)))
nodules_smoker_genes <- unique(na.omit(c(nodules_smoker_clean$SYMBOL, nodules_smoker_clean$gene)))
nodules_neversmoke_genes <- unique(na.omit(c(nodules_neversmoke_clean$SYMBOL, nodules_neversmoke_clean$gene)))
#make a table with the column being the phenotype and the rows being the genes

gene_list <- list(
  bc = bc_genes,
  lcis = lcis_genes,
  dcis = dcis_genes,
  bbd_bn = bbd_bn_genes,
  bbd_pl = bbd_pl_genes,
  bbd_npl = bbd_npl_genes,
  lc_smoker = lc_smoker_genes,
  lc_neversmoke = lc_neversmoke_genes,
  nodules_smoker = nodules_smoker_genes,
  nodules_neversmoke = nodules_neversmoke_genes
)

```




