---
title: "post-gwas"
author: "Christis Nicolaides"
date: "2025-12-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
```{r}

bc <- read.table("./data/raw/bc.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
dcis <- read.table("./data/raw/dcis.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lcis <- read.table("./data/raw/lcis.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_pl <- read.table("./data/raw/bbd_pl.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_npl <- read.table("./data/raw/bbd_npl.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
bbd_bn <- read.table("./data/raw/bbd_bn.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lc_smoker <- read.table("./data/raw/lc_smoker.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
lc_neversmoke <- read.table("./data/raw/lc_neversmoke.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
nodules_smoker <- read.table("./data/raw/nodules_smoker.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)
nodules_neversmoke <- read.table("./data/raw/nodules_neversmoke.sig.txt", header = TRUE, sep = "", fill = TRUE, stringsAsFactors = FALSE)

```


create a column in this format 
10:121586676:A:G	

```{r}
#Download necessary packages

library(tidyverse)

if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(c(
  "GenomicRanges",
  "TxDb.Hsapiens.UCSC.hg38.knownGene",
  "org.Hs.eg.db"
))

library(GenomicRanges)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)

```

```{r}
# new columns 

datasets <- c(
  "bc", 
  "lcis", 
  "dcis", 
  "bbd_bn", 
  "bbd_npl", 
  "bbd_pl", 
  "lc_smoker", 
  "lc_neversmoke",
  "nodules_smoker",
  "nodules_neversmoke"
)

for (d in datasets) {
  df <- get(d) %>% 
    mutate(
      variant_id = str_c(CHROM, GENPOS, ALLELE0, ALLELE1, sep = ":")
    )
  
  assign(d, df)
}

```

VEP annotated datatsets 

```{r}
bc_vep <- read_tsv(
  "./data/raw/bc_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

lcis_vep <- read_tsv(
  "./data/raw/lcis_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

dcis_vep <- read_tsv(
  "./data/raw/dcis_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c")  
)

lc_neversmoke_vep <- read_tsv(
  "./data/raw/bc_annotated.txt",
  col_names = TRUE,
  col_types = cols(.default = "c") 
)

#delete unessesary columns to save memory

bc_vep <- bc_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

lcis_vep <- lcis_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

dcis_vep <- dcis_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

lc_neversmoke_vep <- lc_neversmoke_vep %>%
  dplyr::select('#Uploaded_variation', Location, Allele, Gene, SYMBOL, Existing_variation, REF_ALLELE, UPLOADED_ALLELE, EXON, INTRON)

#create a k
```

```{r}
#make sure whether all variants in bc are in bc_vep

missing_variants <- setdiff(bc$variant_id, bc_vep$`#Uploaded_variation`) #empty

#all represented 

#create a new column on bc which cas the gene symbol based on the data of bc_vep

bc <- bc %>%
  left_join(
    bc_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

lcis <- lcis %>%
  left_join(
    lcis_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  ) %>%
  unique()

dcis <- dcis %>%
  left_join(
    dcis_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  )%>%
  unique()


lc_neversmoke <- lc_neversmoke %>%
  left_join(
    lc_neversmoke_vep %>%
      dplyr::select(`#Uploaded_variation`, SYMBOL, Existing_variation),
    by = c("variant_id" = "#Uploaded_variation")
  )%>%
  unique()

```

```{r}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene

variants_gr <- GRanges(
  seqnames = Rle(paste0("chr", bc$CHROM)),
  ranges = IRanges(start = as.numeric(bc$GENPOS), end = as.numeric(bc$GENPOS)),
  strand = Rle("*"),
  variant_id = bc$variant_id
)

genes_gr <- genes(txdb)
exons_gr <- exons(txdb)
promoters_gr <- promoters(txdb, upstream = 2000, downstream = 200)

hits_genes <- findOverlaps(variants_gr, genes_gr)

bc$gene_id <- NA_character_
bc$gene_id[queryHits(hits_genes)] <-
  mcols(genes_gr)$gene_id[subjectHits(hits_genes)]

hits_exons <- findOverlaps(variants_gr, exons_gr)

bc$in_exon <- FALSE
bc$in_exon[queryHits(hits_exons)] <- TRUE

hits_prom <- findOverlaps(variants_gr, promoters_gr)

bc$in_promoter <- FALSE
bc$in_promoter[queryHits(hits_prom)] <- TRUE




```

